### PR TITLE
Document workaround for aiosqlite savepoints

### PR DESCRIPTION
**Pull request by ziima**
### Description
Update docs with workaround for SAVEPOINT in aiosqlite, based on https://github.com/sqlalchemy/sqlalchemy/discussions/10463

### Checklist
This pull request is:

- [x] A documentation / typographical / small typing error fix
	- Good to go, no issue or tests are needed

### CODE REVIEW

**Review by zzzeek — CHANGES_REQUESTED** [review]

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision c2b2e91887c16b925c071c1210e83341b14130b1 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change c2b2e91887c16b925c071c1210e83341b14130b1: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/scoping.py [hunk 1]
```diff
@@ -470,6 +470,10 @@ def begin_nested(self) -> SessionTransaction:
             with the SQLite driver in order for SAVEPOINT to work
             correctly.
 
+            :ref:`aiosqlite_serializable` - special workarounds required
+            with the SQLite driver in order for SAVEPOINT to work
+            correctly.
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/scoping.py:475]: **Federico Caselli** (CaselIT) wrote:

can we collapse the two, maybe adding `The same holds true for the aiosqlite driver :ref...`

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 2]
```diff
@@ -1904,6 +1905,10 @@ def begin_nested(self) -> SessionTransaction:
             with the SQLite driver in order for SAVEPOINT to work
             correctly.
 
+            :ref:`aiosqlite_serializable` - special workarounds required
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1908]: **Federico Caselli** (CaselIT) wrote:

as the other comment

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Review by sqla-tester — COMMENTED** [review]
**Federico Caselli** (CaselIT) wrote:

code review left on gerrit

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/aiosqlite.py [hunk 3]
```diff
@@ -44,6 +44,36 @@
 aiosqlite extends pysqlite to support async, so we can create our own user-defined functions (UDFs)
 in Python and use them directly in SQLite queries as described here: :ref:`pysqlite_udfs`.
 
+.. _aiosqlite_serializable:
+
+Serializable isolation / Savepoints / Transactional DDL
```
[the remark this answers is not in the document]
Comment (ziima) [reply lib/sqlalchemy/dialects/sqlite/aiosqlite.py]: Fixed

File: lib/sqlalchemy/orm/scoping.py (hunk 1 above)
Comment (ziima) [reply lib/sqlalchemy/orm/scoping.py]: Fixed

File: lib/sqlalchemy/orm/session.py (hunk 2 above)
Comment (ziima) [reply lib/sqlalchemy/orm/session.py]: Fixed

**Comment by ziima** [comment]
I also added notes to the `begin_nested` in the asyncio extension.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 00e3562522eac5f4c454653b52ad1bd15873a835 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 00e3562522eac5f4c454653b52ad1bd15873a835 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/scoping.py [hunk 4]
```diff
@@ -468,7 +468,8 @@ def begin_nested(self) -> SessionTransaction:
 
             :ref:`pysqlite_serializable` - special workarounds required
             with the SQLite driver in order for SAVEPOINT to work
-            correctly.
+            correctly. The same holds true for the aiosqlite driver
+            :ref:`aiosqlite_serializable`.
 
 
         """  # noqa: E501
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/scoping.py:475]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 5]
```diff
@@ -1902,7 +1903,8 @@ def begin_nested(self) -> SessionTransaction:
 
             :ref:`pysqlite_serializable` - special workarounds required
             with the SQLite driver in order for SAVEPOINT to work
-            correctly.
+            correctly. The same holds true for the aiosqlite driver
+            :ref:`aiosqlite_serializable`.
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1908]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Review by sqla-tester — COMMENTED** [review]
**Federico Caselli** (CaselIT) wrote:

thanks for the update

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 0e4b54c54520164bf67930c26731745941ff7118 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 0e4b54c54520164bf67930c26731745941ff7118 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/base.py [hunk 6]
```diff
@@ -273,8 +273,9 @@ def bi_c(element, compiler, **kw):
 .. warning::
 
     SQLite's SAVEPOINT feature is impacted by unresolved
-    issues in the pysqlite driver, which defers BEGIN statements to a greater
-    degree than is often feasible. See the section :ref:`pysqlite_serializable`
+    issues in the pysqlite and aiosqlite driver, which defers BEGIN statements
```
Comment (sqla-tester) [inline lib/sqlalchemy/dialects/sqlite/base.py:276]: **mike bayer** (zzzeek) wrote:

"drivers"

"defer"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/base.py [hunk 7]
```diff
@@ -273,8 +273,9 @@ def bi_c(element, compiler, **kw):
...
+    to a greater degree than is often feasible. See the section
```
Comment (sqla-tester) [inline lib/sqlalchemy/dialects/sqlite/base.py:277]: **mike bayer** (zzzeek) wrote:

"sections"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/base.py [hunk 8]
```diff
@@ -273,8 +273,9 @@ def bi_c(element, compiler, **kw):
...
+    :ref:`pysqlite_serializable` or :ref:`aiosqlite_serializable`
```
Comment (sqla-tester) [inline lib/sqlalchemy/dialects/sqlite/base.py:278]: **mike bayer** (zzzeek) wrote:

"and"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/ext/asyncio/scoping.py [hunk 9]
```diff
@@ -394,6 +394,12 @@ def begin_nested(self) -> AsyncSessionTransaction:
         For a general description of ORM begin nested, see
         :meth:`_orm.Session.begin_nested`.
 
+        .. seealso::
+
+            :ref:`aiosqlite_serializable` - special workarounds required
+            with the SQLite driver in order for SAVEPOINT to work
```
Comment (sqla-tester) [inline lib/sqlalchemy/ext/asyncio/scoping.py:400]: **mike bayer** (zzzeek) wrote:

"with the SQLite asyncio driver"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/ext/asyncio/session.py [hunk 10]
```diff
@@ -979,6 +979,12 @@ def begin_nested(self) -> AsyncSessionTransaction:
         For a general description of ORM begin nested, see
         :meth:`_orm.Session.begin_nested`.
 
+        .. seealso::
+
+            :ref:`aiosqlite_serializable` - special workarounds required
+            with the SQLite driver in order for SAVEPOINT to work
```
Comment (sqla-tester) [inline lib/sqlalchemy/ext/asyncio/session.py:985]: **mike bayer** (zzzeek) wrote:

"with the SQLite asyncio driver"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/scoping.py [hunk 11]
```diff
@@ -468,7 +468,8 @@ def begin_nested(self) -> SessionTransaction:
 
             :ref:`pysqlite_serializable` - special workarounds required
             with the SQLite driver in order for SAVEPOINT to work
-            correctly.
+            correctly. The same holds true for the aiosqlite driver
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/scoping.py:471]: **mike bayer** (zzzeek) wrote:

"For asyncio use cases, see the section :ref:`aiosqlite_serializable`"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 12]
```diff
@@ -1633,7 +1633,8 @@ def __init__(
             .. tip:: When using SQLite, the SQLite driver included through
                Python 3.11 does not handle SAVEPOINTs correctly in all cases
                without workarounds. See the section
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1635]: **mike bayer** (zzzeek) wrote:

"sections"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 13]
```diff
@@ -1633,7 +1633,8 @@ def __init__(
...
-               :ref:`pysqlite_serializable` for details on current workarounds.
+               :ref:`pysqlite_serializable` or :ref:`aiosqlite_serializable`
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1636]: **mike bayer** (zzzeek) wrote:

"and"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 14]
```diff
@@ -1902,7 +1903,8 @@ def begin_nested(self) -> SessionTransaction:
 
             :ref:`pysqlite_serializable` - special workarounds required
             with the SQLite driver in order for SAVEPOINT to work
-            correctly.
+            correctly. The same holds true for the aiosqlite driver
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1906]: **mike bayer** (zzzeek) wrote:

"For asyncio use cases, see the section :ref:`aiosqlite_serializable`"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Review by sqla-tester — COMMENTED** [review]
**mike bayer** (zzzeek) wrote:

code review left on gerrit

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Comment by ziima** [comment]
Fixed, thanks for suggestions.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 0516c205d2c4c046d449b5684c13a06f813dabe1 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 0516c205d2c4c046d449b5684c13a06f813dabe1 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/base.py [hunk 15]
```diff
@@ -273,8 +273,9 @@ def bi_c(element, compiler, **kw):
 .. warning::
 
     SQLite's SAVEPOINT feature is impacted by unresolved
-    issues in the pysqlite driver, which defers BEGIN statements to a greater
-    degree than is often feasible. See the section :ref:`pysqlite_serializable`
+    issues in the pysqlite and aiosqlite drivers, which defer BEGIN statements
```
Comment (sqla-tester) [inline lib/sqlalchemy/dialects/sqlite/base.py:276]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/base.py [hunk 16]
```diff
@@ -273,8 +273,9 @@ def bi_c(element, compiler, **kw):
...
+    to a greater degree than is often feasible. See the sections
```
Comment (sqla-tester) [inline lib/sqlalchemy/dialects/sqlite/base.py:277]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/dialects/sqlite/base.py [hunk 17]
```diff
@@ -273,8 +273,9 @@ def bi_c(element, compiler, **kw):
...
+    :ref:`pysqlite_serializable` and :ref:`aiosqlite_serializable`
```
Comment (sqla-tester) [inline lib/sqlalchemy/dialects/sqlite/base.py:278]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/ext/asyncio/scoping.py [hunk 18]
```diff
@@ -394,6 +394,12 @@ def begin_nested(self) -> AsyncSessionTransaction:
         For a general description of ORM begin nested, see
         :meth:`_orm.Session.begin_nested`.
 
+        .. seealso::
+
+            :ref:`aiosqlite_serializable` - special workarounds required
+            with the SQLite asyncio driver in order for SAVEPOINT to work
```
Comment (sqla-tester) [inline lib/sqlalchemy/ext/asyncio/scoping.py:400]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/ext/asyncio/session.py [hunk 19]
```diff
@@ -979,6 +979,12 @@ def begin_nested(self) -> AsyncSessionTransaction:
         For a general description of ORM begin nested, see
         :meth:`_orm.Session.begin_nested`.
 
+        .. seealso::
+
+            :ref:`aiosqlite_serializable` - special workarounds required
+            with the SQLite asyncio driver in order for SAVEPOINT to work
```
Comment (sqla-tester) [inline lib/sqlalchemy/ext/asyncio/session.py:985]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/scoping.py [hunk 20]
```diff
@@ -468,7 +468,8 @@ def begin_nested(self) -> SessionTransaction:
 
             :ref:`pysqlite_serializable` - special workarounds required
             with the SQLite driver in order for SAVEPOINT to work
-            correctly.
+            correctly. For asyncio use cases, see the section
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/scoping.py:471]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 21]
```diff
@@ -1632,8 +1632,9 @@ def __init__(
 
             .. tip:: When using SQLite, the SQLite driver included through
                Python 3.11 does not handle SAVEPOINTs correctly in all cases
-               without workarounds. See the section
-               :ref:`pysqlite_serializable` for details on current workarounds.
+               without workarounds. See the sections
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1635]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 22]
```diff
@@ -1632,8 +1632,9 @@ def __init__(
...
+               :ref:`pysqlite_serializable` and :ref:`aiosqlite_serializable`
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1636]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

File: lib/sqlalchemy/orm/session.py [hunk 23]
```diff
@@ -1902,7 +1903,8 @@ def begin_nested(self) -> SessionTransaction:
 
             :ref:`pysqlite_serializable` - special workarounds required
             with the SQLite driver in order for SAVEPOINT to work
-            correctly.
+            correctly. For asyncio use cases, see the section
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/session.py:1906]: **Federico Caselli** (CaselIT) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Review by sqla-tester — COMMENTED** [review]
**Federico Caselli** (CaselIT) wrote:

code review left on gerrit

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

great, thank you for the updates!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

will wait for re-verification

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4907 has been **merged**. Congratulations! :)

**Pull request closed without merging**